server/src-lib/Hasura/GraphQL/Execute/RemoteJoins.hs

```diff
@@ -164,54 +159,48 @@ emptyResp = GQRespValue OJ.empty VB.empty
 -- | Insert at path, index the value in the larger structure.
 insertBatchResults ::
      GQRespValue
-  -- ^ Original hasura response with accumulated remote responses
+  -- ^ Original hasura response
   -> Batch
   -> GQRespValue
+  -- ^ Remote response
   -> GQRespValue
+  -- ^ Original hasura response with accumulated remote responses and errors
 insertBatchResults accumResp Batch{..} remoteResp =
   -- It's not clear what to do about errors here, or when to short-circuit so
   -- try to do as much computation as possible for now:
-  case remoteDataE of
-    Left err -> appendJoinError accumResp err
-    Right remoteData0 -> do
+  case orderedTopLevelRemoteSelections of
+    Nothing -> appendJoinError accumResp $ 
+      "did not find expected hasura alias prefex in all remote results: " <>
+      fromString (show $ _gqRespData remoteResp)
+    Just remoteData0 -> do
       case inHashmap batchRelFieldPath accumData0 remoteData0 of
-        Left err       -> appendJoinError accumRespWithRemoteErrs err
-        Right (val, _) -> set gqRespData val accumRespWithRemoteErrs
+        Left err -> appendJoinError accumRespWithRemoteErrs err
+        -- TODO assert something about _remainingRemote?
+        Right (val, _remainingRemote) -> set gqRespData val accumRespWithRemoteErrs
```

Comment:
Ideally `_remainingRemotes` should be an empty list since we expect all the remotes to have been joined.